### DISCUSSION TITLE
why the speed of log10 function is much slower than log?

### DISCUSSION THREAD

**Discussion by l1t1** [ClickHouse/ClickHouse#49591] [Q&A] [answered]
create table t engine=Memory as SELECT number::int i FROM numbers(1,1000000000)t;
SELECT avg(i) FROM t group by round(log(i)); ---2.7s
SELECT avg(i) FROM t group by round(log10(i)); --5.7s

**Comment by alexey-milovidov** [comment, marked as answer]
They use different underlying implementations.
log - [FastOps](https://github.com/ClickHouse/fastops/tree/1460583af7d13c0e980ce46aec8ee9400314669a)
log10 - GLibc

It's possible to add the support for log10 into FastOps and improve its performance,
or implement it with log and one multiplication by a constant - it will be almost as fast but will introduce a small loss of precision.

**Comment by l1t1** [comment]
thank you
